sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -379,15 +390,13 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
       case _: String => StringType
     }
 
-    val columnEquals = df.sparkSession.sessionState.analyzer.resolver
-    val projections = df.schema.fields.map { f =>
-      val shouldReplace = cols.exists(colName => columnEquals(colName, f.name))
```

Comment:
Previously, we use string match directly, which means `df.na.replace("col.A", ...)` works. Now, we parse the column name as SQL syntax, which may break `df.na.replace("col.A", ...)`.

I think the new behavior makes more sense since it makes SQL and DataFrame API more consistent about column names. It also allows qualified column names which is useful for joined dataframes. Let's add an item in the migration guide to describe the behavior change.

also cc @HyukjinKwon @srowen

Reply:
Yea, I think we need this change for matching the behaivour of the other functions in this class, e.g., `drop`.

Reply:
Added note in the migration guide, please review.